### PR TITLE
feat: Implementing the linking between documents and activities

### PR DESCRIPTION
**Pull request by yassine04e**
In this PR we link finalise the linking between existing documents to activities.

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt [hunk 1]
```diff
@@ -89,57 +101,57 @@ class TravelActivityScreenCopy {
...
+    composeTestRule.setContent {
+      ActivityItem(activity, {}, LocalContext.current, mockDocumentViewModel)
+    }
     composeTestRule.waitForIdle()
     composeTestRule.onNodeWithTag("activityItem").assertIsDisplayed()
     composeTestRule.onNodeWithTag("activityItem").assertTextContains(activity.title)
     composeTestRule.onNodeWithTag("activityItem").assertTextContains(activity.location.name)
     composeTestRule.onNodeWithTag("activityItem").assertTextContains("1/1/1970")
-    composeTestRule.onNodeWithTag("extraDocumentButton").assertIsDisplayed().performClick()
+    // composeTestRule.onNodeWithTag("extraDocumentButton").assertIsDisplayed().performClick()
```
Comment (RemIsMyWaifuu) [inline app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt:113]:
```suggestion
```
I suggest removing the extra comment

File: app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt [hunk 2]
```diff
@@ -89,57 +101,57 @@ class TravelActivityScreenCopy {
...
+  // "https://img.yumpu.com/30185842/1/500x640/afps-attestation-de-formation-aux-premiers-secours-programme-.jpg",
+  //
+  // "https://wallpapercrafter.com/desktop6/1606440-architecture-buildings-city-downtown-finance-financial.jpg",
+  //
+  // "https://assets.entrepreneur.com/content/3x2/2000/20151023204134-poker-game-gambling-gamble-cards-money-chips-game.jpeg")
+  //    composeTestRule.setContent {
+  //      ActivityItem(activity, {}, LocalContext.current, listOf(images[0], images[1]))
+  //    }
+  //    composeTestRule.waitForIdle()
+  //  }
```
Comment (RemIsMyWaifuu) [inline app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt:148]:
```suggestion
```
Please remove commented lines

File: app/src/main/java/com/github/se/travelpouch/model/activity/ActivityRepositoryFirebase.kt [hunk 3]
```diff
@@ -178,9 +181,24 @@ class ActivityRepositoryFirebase(private val db: FirebaseFirestore) : ActivityRe
...
+                    title = it["title"] as String,
+                    fileFormat = DocumentFileFormat.valueOf(it["fileFormat"] as String),
+                    fileSize = it["fileSize"] as Long,
+                    addedByEmail = it["addedByEmail"] as String?,
+                    addedByUser = it["addedByUser"] as DocumentReference?,
+                    addedAt = it["addedAt"] as Timestamp,
+                    visibility = DocumentVisibility.valueOf(it["visibility"] as String))
+              }
+              ?.toList() ?: emptyList()
+      // documentsNeededData?.map { (key, value) -> key as String to value as Int }?.toMap()
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/activity/ActivityRepositoryFirebase.kt:201]: remove unused comment please
```suggestion
```

File: app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt [hunk 4]
```diff
@@ -79,21 +85,62 @@ class DocumentRepositoryFirestore(
...
+  //        onSuccess()
+  //      } else {
+  //        task.exception?.let { e ->
+  //          Log.e("DocumentRepositoryFirestore", "Error deleting document", e)
+  //          onFailure(e)
+  //        }
+  //      }
+  //    }
+  //  }
+
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt:104]:
```suggestion
```

Please remove commented out lines

File: app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentPreview.kt [hunk 5]
```diff
@@ -109,12 +133,30 @@ fun DocumentPreview(documentViewModel: DocumentViewModel, navigationActions: Nav
...
                   },
                   modifier = Modifier.testTag("deleteButton")) {
                     Icon(imageVector = Icons.Default.Delete, contentDescription = "Delete Document")
                   }
+
+              IconButton(
+                  onClick = {
+                    if (activityViewModel.activities.value.isEmpty()) {
+                      Toast.makeText(context, "No Activities to available", Toast.LENGTH_LONG)
+                          .show()
```
Comment (RemIsMyWaifuu) [inline app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentPreview.kt:152]: Could you please change the toast to "Cannot link image if there are no activities" ?

**Review by RemIsMyWaifuu — APPROVED** [review]
Overall, solid work. Some cleanup needs to be made but in terms of implementation I have already made some UI changes by my own commit.
It works when I used it. Its well scoped, it's not all we would have wanted it to be but for the amount of time we have left this is an impressive piece of work!

File: app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt (hunk 1 above)
Comment (yassine04e) [reply app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt:113]: Done

File: app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt (hunk 2 above)
Comment (yassine04e) [reply app/src/androidTest/java/com/github/se/travelpouch/ui/dashboard/TravelActivityScreen.kt:148]: Done

File: app/src/main/java/com/github/se/travelpouch/model/activity/ActivityRepositoryFirebase.kt (hunk 3 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/activity/ActivityRepositoryFirebase.kt:201]: Done

File: app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt (hunk 4 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt:104]: Done

File: app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentPreview.kt (hunk 5 above)
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentPreview.kt:150]: Done

File: app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentPreview.kt [hunk 6]
```diff
@@ -145,5 +187,82 @@ fun DocumentPreview(documentViewModel: DocumentViewModel, navigationActions: Nav
...
+      ) {
+        Column(
+            modifier =
+                Modifier.fillMaxWidth(1f)
+                    .background(MaterialTheme.colorScheme.background)
+                    .testTag("activitiesDialog"),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.SpaceEvenly) {
+              Text(
+                  "Link to what activities",
```
[the remark this answers is not in the document]
Comment (yassine04e) [reply app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentPreview.kt:201]: Done. I am sorry. It was done between 3 and 5 AM. Mistakes were done

**Pull request merged**